src/py/flwr/client/app.py

```diff
@@ -365,7 +367,23 @@ def _load_app() -> Flower:
                 app: Flower = load_flower_callable_fn()
 
                 # Handle task message
-                out_message = app(message=message, context=context)
+                try:
+                    out_message = app(message=message, context=context)
+                except Exception as ex:  # pylint: disable=broad-exception-caught
+                    log(ERROR, "FlowerCallable raised an exception", exc_info=ex)
+
+                    # Don't update/change RunState
+                    # Return empty Message
+
+                    error_out_message = Message(
+                        metadata=message.metadata,
+                        message=RecordSet(),
+                    )
```

Comment:
Though we don't send any error report for now, I think we can set the `task_type="error"` to indicate this is an error message, instead of sending an empty message.

Reply:
I replaced this with the new error mechanism